src/Compilers/CSharp/Portable/FlowAnalysis/PreciseAbstractFlowPass.cs

```diff
@@ -1264,131 +1259,55 @@ public override BoundNode VisitIndexerAccess(BoundIndexerAccess node)
         {
             var method = GetReadMethod(node.Indexer);
             VisitReceiverBeforeCall(node.ReceiverOpt, method);
-            VisitArguments(node.Arguments, node.ArgumentRefKindsOpt, method, node.ArgsToParamsOpt, node.Expanded);
+            VisitArguments(node.Arguments, node.ArgumentRefKindsOpt, method);
             if (_trackExceptions && (object)method != null) NotePossibleException(node);
             if ((object)method != null) VisitReceiverAfterCall(node.ReceiverOpt, method);
             return null;
         }
 
         public override BoundNode VisitEventAssignmentOperator(BoundEventAssignmentOperator node)
         {
-            VisitReceiverOfEventAssignmentAsRvalue(node);
+            VisitRvalue(node.ReceiverOpt);
             VisitRvalue(node.Argument);
             if (_trackExceptions) NotePossibleException(node);
             return null;
         }
 
-        protected virtual void VisitReceiverOfEventAssignmentAsRvalue(BoundEventAssignmentOperator node)
+        private void VisitArguments(ImmutableArray<BoundExpression> arguments, ImmutableArray<RefKind> refKindsOpt, MethodSymbol method)
         {
-            VisitRvalue(node.ReceiverOpt);
-        }
+            Debug.Assert(!(this is NullableWalker));  // Should call NullableWalker.VisitArguments instead, for null warnings.
```

Comment:
Consider making this a virtual method and overriding in NullableWalker to achieve this.

Reply:
Done.